### ISSUE TITLE
False positive when ignoring the yield type of an async context manager

### ISSUE THREAD

**Issue by Apakottur** [microsoft/pyright#11020]
**Describe the bug**
Consider the following code:
```python
# test.py
import asyncio
from contextlib import asynccontextmanager
from typing import AsyncIterator

@asynccontextmanager
async def manager() -> AsyncIterator[int]:
    yield 1

async def main() -> None:
    async with (
        manager(),
    ):
        print("Hello!")

asyncio.run(main())
```

Running `pyright test.py` gives:
```text
./test.py
  ./test.py:13:16 - error: Object of type "tuple[_AsyncGeneratorContextManager[int, None]]" cannot be used with "async with" because it does not correctly implement __aenter__
    Attribute "__aenter__" is unknown (reportGeneralTypeIssues)
  ./test.py:13:16 - error: Object of type "tuple[_AsyncGeneratorContextManager[int, None]]" cannot be used with "with" because it does not correctly implement __aexit__
    Attribute "__aexit__" is unknown (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 informations
```

The code is valid and prints "Hello!" as expected, but Pyright seems to have an issue with the context manager.
Am I missing something or is this a bug?

I'm using `pyright==1.1.406`.

**Comment by erictraut** [comment]
Thanks for reporting the issue. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.407
